app/socket-apis/get-points-of-ids.js

```diff
@@ -0,0 +1,64 @@
+// https://github.com/EnCiv/civil-pursuit/issues/203
+const Points = require('../models/points')
+const { ObjectId } = require('mongodb')
+
+async function getPointsOfIds(ids, callback) {
+  const cbFailure = errorMsg => {
+    if (errorMsg) console.error(errorMsg)
+    if (callback) callback({ points: [], myWhys: [] }) // Return empty arrays on error
+  }
+
+  // Verify user is logged in by checking synuser in ids
+  const synuser = ids.find(id => typeof id === 'object' && id.synuser)
```

Comment:
This is how to verify the user is logged in,
```
 // Verify user is logged in.
  if (!this.synuser || !this.synuser.id) {
    return cbFailure('User is not logged in.')
  }
```
If there is trouble using `this` as a variable it's a problem with calling the function.